flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniClusterConfiguration.java

```diff
@@ -206,6 +207,11 @@ public Builder setHaServices(MiniCluster.HaServices haServices) {
             return this;
         }
 
+        public Builder withRandomPorts() {
```

Comment:
Would it make sense to use random ports by default and have a builder method for the static port allocation? That's IMO what we want in most scenarios.

eg.
```
withStaticPorts(...)
```

Reply:
The MiniCluster is in a weird spot where it is not a pure testing utility.
If you consider it to just be a single JVM Flink deployment, then it makes perfect sense for it to use the usual ports.

I'm wary of changing the default MiniCluster behavior because it may break plenty of stuff for users.